server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1966,13 +1966,13 @@ public Response bind(
             throw new BadRequestException(i18n.tr("Owner has autobind disabled."));
         }
 
-        List<PoolQuantity> dryRunPools = new ArrayList<PoolQuantity>();
+        List<PoolQuantityDTO> dryRunPools = new ArrayList<PoolQuantityDTO>();
 
         try {
             consumerBindUtil.validateServiceLevel(consumer.getOwner(), serviceLevel);
             dryRunPools = entitler.getDryRun(consumer, serviceLevel);
             if (dryRunPools == null) {
-                return new ArrayList<PoolQuantity>();
+                return new ArrayList<PoolQuantityDTO>();
```

Comment:
We can refactor this block a bit such that we're not creating the collection only to throw it out and, in some cases, recreate it.

Reply:
What I was referring to here was the creation of the arraylist. We could do something like:

```
List<PoolQuantity> dryRunPools = null;

try {
    ...
    dryRunPools = entitler.getDryRun(...);
}
// more catch stuff here

if (dryRunPools != null) {
  // do translation here
}
else {
  return Collections.<PoolQuantityDTO>emptyList();
}
```